README.md

```diff
@@ -371,6 +371,8 @@ To fix the incorrect port, start the Docker service with the `--userland-proxy=f
 
 Use the `PORT` environment variable to start the server on the a different port, .e.g. `docker run -e "PORT=34198"`. This changes the source port on the packets used for port detection. `-p 34198:34197` works fine for private servers, but the server browser detects the wrong port.
```

Comment:
See. This was already possible but I was just not aware of it. Closing this and adding some more doc about it.

Reply:
Bruh....

It doesn’t work for me, I’m not launching it with docker directly, I’m using docker-compose

Did you even read the whole sentence? 

> but the server browser detects the wrong port.

This is exactly my problem with it, my friends can’t join the server because of this

Reply:
Honestly why you against something that improves the docker image and the work was done for you?

Reply:
You can also set the PORT environment variable with docker-compose. Your patch does not change anything but adding another environment variable to change the existing PORT variable.

Reply:
Setting port does not start the server with the wanted port... try it and see for yourself. It’s get’s overridden by the docker image...

Reply:
If you want the game to work on multiplayer mode and be discoverable on the factorio browser you need to override the starting port

Reply:
This works for me.
```
docker run -e PORT=12345 factoriotools/factorio
...
   0.773 Info UDPSocket.cpp:27: Opening socket at (IP ADDR:({0.0.0.0:12345}))
...
```

Reply:
So why doesn’t this work with docker-compose? 
If this does work them something else is the matter here

Reply:
Did you uncomment the environment section? This works for me like it should.

Reply:
I used a docker-compose.yaml file, I did not use the dockerfile, I set the PORT variable correctly in docker-compose but I suspect that the variable was overwritten via the Dockerfile because it was using the hardcoded port

Reply:
It is not hardcoded there. I tested it.

Reply:
Show me with that working using docker-compose.... because that’s what I’m using as I’ve said before, the main thing is to be able to join a server from inside the game in the browser window.

I’m hosting 4 factorio servers on a single server with my updated imageand all can be joined via the server browser instead of local / direct connections.

Reply:
```
version: '2'
services:
  factorio:
    build: .
    ports:
     - "34197:34197/udp"
     - "27015:27015/tcp"
    volumes:
     - /opt/factorio:/factorio
#    environment:
#      - PUID=1000
#      - PGID=1000
#      - UPDATE_MODS_ON_START=true
#      - USERNAME=FactorioUsername
#      - TOKEN=FactorioToken
      - PORT=12345
```